File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/release_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class ReleaseCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(ReleaseCC.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.NZ}

File: golang/uPIMulator/src/linker/kernel/instruction/cc/release_cc.go
package cc

import (
	"errors"
)

type ReleaseCc struct {
	condition Condition
}

func (this *ReleaseCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		NZ: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *ReleaseCc) Condition() Condition {
	return this.condition
}
